### PR TITLE
feat: Port `UserListItem` to functional style

### PR DESCRIPTION
**Pull request by CuriousCorrelation**
# Description

PR ports `UserListItem` from [scenes/Settings/components/UserListItem.tsx](https://github.com/outline/outline/blob/main/app/scenes/Settings/components/UserListItem.tsx) to functional style.

Also adds translations for "Invited" here

https://github.com/outline/outline/blob/04f5b08ba18c50a670cdab946b49f03dfc412d71/app/scenes/Settings/components/UserListItem.tsx#L28-L34

### CODE REVIEW

File: app/scenes/Settings/components/UserListItem.tsx [hunk 1]
```diff
@@ -13,34 +14,31 @@ type Props = {
...
+          {user.email ? `${user.email} · ` : undefined}
+          {user.lastActiveAt ? (
+            <>
+              Active <Time dateTime={user.lastActiveAt} /> ago
+            </>
+          ) : (
+            t("Invited")
+          )}
+          {user.isAdmin && <Badge primary={user.isAdmin}>Admin</Badge>}
+          {user.isSuspended && <Badge>Suspended</Badge>}
```
Comment (tommoor) [inline app/scenes/Settings/components/UserListItem.tsx:35]: Needs translating

File: app/scenes/Settings/components/UserListItem.tsx [hunk 2]
```diff
@@ -13,34 +14,31 @@ type Props = {
...
+        <>
+          {user.email ? `${user.email} · ` : undefined}
+          {user.lastActiveAt ? (
+            <>
+              Active <Time dateTime={user.lastActiveAt} /> ago
+            </>
+          ) : (
+            t("Invited")
+          )}
+          {user.isAdmin && <Badge primary={user.isAdmin}>Admin</Badge>}
```
Comment (tommoor) [inline app/scenes/Settings/components/UserListItem.tsx:34]: Needs translating

File: app/scenes/Settings/components/UserListItem.tsx [hunk 3]
```diff
@@ -13,34 +14,31 @@ type Props = {
...
+  return (
+    <ListItem
+      title={<Title>{user.name}</Title>}
+      image={<Avatar src={user.avatarUrl} size={32} />}
+      subtitle={
+        <>
+          {user.email ? `${user.email} · ` : undefined}
+          {user.lastActiveAt ? (
+            <>
+              Active <Time dateTime={user.lastActiveAt} /> ago
```
Comment (tommoor) [inline app/scenes/Settings/components/UserListItem.tsx:29]: Needs translating, you'll need to use the `<Trans>` component with this one to insert the <Time> component in the right place… see other examples in the codebase.

**Review by tommoor — COMMENTED** [review]
You spotted one string but missed the other three 😉

File: app/scenes/Settings/components/UserListItem.tsx (hunk 3 above)
Comment (CuriousCorrelation) [reply app/scenes/Settings/components/UserListItem.tsx:29]: Found exactly what I was looking for https://github.com/outline/outline/blob/04f5b08ba18c50a670cdab946b49f03dfc412d71/app/scenes/Document/components/Notices.tsx#L23-L26

**Comment by CuriousCorrelation** [comment]
Thanks for the review! I missed those even after including them in my PR... :disappointed: 

I'll be extra careful next time

**Comment by CuriousCorrelation** [comment]
I think I remember reading it somewhere that sometimes husky completely skips hooks if there's something wrong with `yarn.lock` file. Removed it and ran `yarn` to rebuild everything, re-ran husky, but still nothing. Tried manually running the hooks. Nothing again.

What worked was running `yarn (install)` followed `yarn i18next [long/path]/UserListItem.tsx` which finally generated the file, of course had to merge diffs to get back the original with updates appended.

Seems like the problem was a combination of these two. Might need to do some further digging to make sure.

**Comment by tommoor** [comment]
The file still doesn't have all the new translations in it though 🤔 . Not sure why husky wouldn't install

**Comment by CuriousCorrelation** [comment]
Very odd indeed :monocle_face: . Tried a lot of things to diagnose the issue. So far it seems husky is installed correctly. Hooks are being triggered as expected. Even compiling from source works just the same. Manually running hooks with verbose shows this

``` sh
⚠ Running tasks for staged files...
  ❯ lint-staged.config.mjs — 1 file
    ❯ **/*.[tj]s?(x) — 1 file
       eslint /home/atis/Projects/outline/app/scenes/Settings/components/UserListItem.tsx --fix [SIGINT]
      ◼ yarn build:i18n
      ◼ git add shared/i18n/locales/en_US/translation.json
    ↓ (yarn.lock|package.json) — no files [SKIPPED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...
```

concluding with

``` sh
✔ Preparing lint-staged...
✔ Running tasks for staged files...
✔ Applying modifications from tasks...
✔ Cleaning up temporary files...
```

beside that `yarn` shows no issue. `node` is up to date. All packages are installed correctly. Editor isn't overriding defaults.

Could this be because translations for these words already exist on main?

https://github.com/outline/outline/blob/9cafe75bf2032727aa0fd71d1af7a5928d9cf793/shared/i18n/locales/en_US/translation.json#L373

https://github.com/outline/outline/blob/9cafe75bf2032727aa0fd71d1af7a5928d9cf793/shared/i18n/locales/en_US/translation.json#L570

https://github.com/outline/outline/blob/9cafe75bf2032727aa0fd71d1af7a5928d9cf793/shared/i18n/locales/en_US/translation.json#L147

https://github.com/outline/outline/blob/9cafe75bf2032727aa0fd71d1af7a5928d9cf793/shared/i18n/locales/en_US/translation.json#L371

I could try a couple more things if this isn't it.

**Comment by tommoor** [comment]
> Could this be because translations for these words already exist on main?

Must be it, I didn't think there was a chance that <Trans> one would exist already

**Comment by CuriousCorrelation** [comment]
Same here! That's why checking translations didn't even cross my mind. I figured that out only after compiling husky from source and adding in some good 'ol `console.log`s.

I'll rollback previous commit. PR should be good to go after

File: app/scenes/Settings/components/UserListItem.tsx [hunk 4]
```diff
@@ -31,8 +31,8 @@ const UserListItem = ({ user, showMenu }: Props) => {
           ) : (
             t("Invited")
           )}
-          {user.isAdmin && <Badge primary={user.isAdmin}>t("Admin")</Badge>}
-          {user.isSuspended && <Badge>t("Suspended")</Badge>}
+          {user.isAdmin && <Badge primary={user.isAdmin}>{t("Admin")}</Badge>}
```
Comment (tommoor) [inline app/scenes/Settings/components/UserListItem.tsx:34]: @CuriousCorrelation Note I had to fix these lines

Comment (CuriousCorrelation) [reply app/scenes/Settings/components/UserListItem.tsx:34]: Thank you, noted

**Review by tommoor — APPROVED** [review]

**Pull request merged**